### PR TITLE
Documentation for Linear search

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added the documentation for linear search 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Closes https://github.com/TesseractCoding/NeoAlgo-Docs/issues/13

### CODE REVIEW

**Comment by SAY-droid427** [comment]
@HarshCasper I cannot understand why the checks are failing. Could you please help me in this?

File: docs/searching-algorithms/linear-search.md [hunk 1]
```diff
@@ -3,4 +3,161 @@ title: Linear Search
...
+
+In linear search, no extra space is used to store anything. So the space complexity is O(1).
+
+### Time Complexity: O(n)
+### Time Complexity: O(1)
+
+
+
+### Credits
+1. [Rajiv Singh]() for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Linear_Search.cpp)
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.md]: Change `[Rajiv Singh]()` to `[Rajiv Singh](https://github.com/iamrajiv)` and the CI will be fixed

**Comment by HarshCasper** [comment]
> @HarshCasper I cannot understand why the checks are failing. Could you please help me in this?

I'm fixing the CI for you. Don't worry

**Comment by HarshCasper** [comment]
Hi @SAY-droid427 

The CI is passing now 😄

File: docs/searching-algorithms/linear-search.mdx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+# Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="js"
+  values={[
+    { label: 'JavaScript', value: 'js', },
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:75]: It should be `{ label: 'C++', value: 'cpp', },`

File: docs/searching-algorithms/linear-search.mdx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    { label: 'Python', value: 'py', },
+    { label: 'Java', value: 'java', },
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:77]: We don't have Java here. Remove it up

File: docs/searching-algorithms/linear-search.mdx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+Enter element to search 3
+```
+### Output :
+
+```
+Found at 3
+
+```
+
+# Time & Space Complexity
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:167]: Keep it `Complexity Analysis`

File: docs/searching-algorithms/linear-search.mdx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+Linear search gives linear access to the elements. In the best case, when the element to be searched is present in the first position, then only one iteration will be required. In the worst case when the element to be searched is present in the last position, linear search will have to go through all the elements. Therefore its time complexity is O(n).
+
+In linear search, no extra space is used to store anything. So the space complexity is O(1).
+
+Time Complexity: O(n)
+Time Complexity: O(1)
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:174]: It should be `Time & Space` complexity not Time and Time complexity. Also use bulletpoints here

File: docs/searching-algorithms/linear-search.mdx [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+
+
+### Credits
+1. [Rajiv Singh](https://github.com/iamrajiv) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Linear_Search.cpp)
+
+2. [Ricardo Prins](https://github.com/ricardoprins) for the [Python implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/search/Linear_Search.py)
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:181]: I don't think they actually wrote it. Both were added here: https://github.com/TesseractCoding/NeoAlgo/pull/56

Kindly give the apt credits for that

File: docs/searching-algorithms/linear-search.mdx [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+ Key = 5 does not match the arr[2]
+ 
+ Next key is compared to arr[3] = 7
+ Key = 5 does not match the arr[3]
+
+ The end of the array is reached. 
+ The key is not present in the array
+ -1 is returned indicating the absence of the key in array 
+
+```
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:65]: Use a smaller example that is easy to consume. Be minimal :)

File: docs/searching-algorithms/linear-search.mdx [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+# Algorithm
+
+In linear search, we follow the following algorithm:
+
+1. Starting from the leftmost element of the array, compare each element with the key element
+
+2. If the element compared is same as the key, then return the index of the element which matched the key
+
+3. If the key does not match any element scanned in the array, then return -1
+
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:19]: Use bulleted points

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you look at these

**Comment by SAY-droid427** [comment]
@HarshCasper  I am sorry I again faced that error saying:
Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
I really have faced this issue a few times

**Comment by HarshCasper** [comment]
> @HarshCasper I am sorry I again faced that error saying:
> Updates were rejected because the remote contains work that you do
> hint: not have locally. This is usually caused by another repository pushing
> hint: to the same ref. You may want to first integrate the remote changes
> hint: (e.g., 'git pull ...') before pushing again.
> I really have faced this issue a few times
> I have created another PR.

No worries :) 

Your other PR is failing checks as well. Can you debug it?

**Pull request closed without merging**